# üåø Cladevo

**_Create phylogenetic trees quickly and efficiently with minimal effort!_**

## **What is Cladevo?**

### Cladevo is a lightweight, open-access tool for constructing phylogenetic trees. It enables rapid analysis of genetic data, designed for researchers, educators, and students who need a simple and accessible solution without relying on heavy bioinformatics software.
 
### Example dendrogram:
![Logo](cladevo/static/images/tree11.png)

## Overview

Cladevo urilizes UPGMA (Unweighted Pair Group Method with Arithmetic Mean) and Neighbor-Joining (NJ) algorithms. It allows generating dendrograms quickly based on flexible data input methods, including FASTA files, Excel files, or manual DNA sequence input.

This tool is ideal for anybody who needs to perform phylogenetic analysis avoiding steep learning curve typical for this kind of software. Scientists can use Cladevo for preliminary data exploration before transitioning to more advanced bioinformatics tools like `MEGA` or `DarWin`, while students can gain hands-on experience with phylogenetic tree construction and aquire basic understanding of the techniques in one place.


## ‚ú® Features

- üå≥ **Phylogenetic Tree Construction**: Neighbor-Joining (NJ) & UPGMA support  
- üìÇ **Flexible Input**: FASTA, Excel, or manual sequences  
- ‚öôÔ∏è **Automated Processing**: sequence alignment, dissimilarity matrix clustering, and dendrogram creation abstracted out
- üß¨ **Molecular Marker Support**: real research data input support (SSR genotyping, PCR, gel electrophoresis)
- ‚ö° **Fast & Accessible**: no installation required when hosted online, Docker-ready for local use  
- üìö **Educational Resources**: built-in theory section on UPGMA & NJ algorithms

## **Tech Stack**

| Layer | Technologies |
|-------|--------------|
| **Backend** | ![Python](https://img.shields.io/badge/Python-3776AB?style=flat&logo=python&logoColor=white) ![Django](https://img.shields.io/badge/Django-092E20?style=flat&logo=django&logoColor=white) |
| **Frontend** | ![HTML](https://img.shields.io/badge/HTML-E34F26?style=flat&logo=html5&logoColor=white) ![CSS](https://img.shields.io/badge/CSS-1572B6?style=flat&logo=css3&logoColor=white) ![JavaScript](https://img.shields.io/badge/JavaScript-F7DF1E?style=flat&logo=javascript&logoColor=black) |
| **Tools / DevOps** | ![Git](https://img.shields.io/badge/Git-F05032?style=flat&logo=git&logoColor=white) ![Linux](https://img.shields.io/badge/Linux-FCC624?style=flat&logo=linux&logoColor=black) ![Docker](https://img.shields.io/badge/Docker-2496ED?style=flat&logo=docker&logoColor=white) |
| **Bioinformatics Tools** | ![Pandas](https://img.shields.io/badge/Pandas-150458?style=flat&logo=pandas&logoColor=white) ![Biopython](https://img.shields.io/badge/Biopython-3673A5?style=flat&logo=biopython&logoColor=white) ![Matplotlib](https://img.shields.io/badge/Matplotlib-11557C?style=flat&logo=matplotlib&logoColor=white) |

---

## Installation & Setup

#### Cladevo is currently unavailable via the web but can be downloaded and installed using the Docker image, available here:[Cladevo](https://hub.docker.com/r/highlander95/cladevo-app).

#### For installation and running instructions, please refer to the Docker documentation, which provides detailed guidance on how to install Docker and run Docker images on your local machine: [Docker Installation and Running Guide](https://docs.docker.com/get-docker/)

## How to Use Cladevo

**_Navigate to `Build Your Tree` or `Get Started` section. Select Algorithm: Choose between UPGMA and Neighbor-Joining. Input data in one of the following formats:_**
* Manual DNA sequence data input.
  ![Login GIF](assets/manual.gif)
* FASTA file DNA sequence data input.
  ![Login GIF](assets/fasta.gif)
* EXCEL file DNA sequence data input.
  ![Login GIF](assets/excel.gif)

* Download Results: Save the tree visualization for further analysis.

## Educational Resources

Cladevo provides learning materials to help users understand the underlying algorithms:

* Theory Section: Learn about UPGMA and Neighbor-Joining.

* How to Use Section: Watch video tutorials on using Cladevo effectively.

(The sections are available inside the app after installation.)

## License

This project is released under the [MIT License](https://opensource.org/license/MIT), allowing users to freely use, modify, and distribute Cladevo's code while retaining the original copyright notice. However, Cladevo's frontend is built using an [HTML5 UP](https://html5up.net/) template, which falls under the [Creative Commons Attribution 3.0 License](https://html5up.net/license)‚Äîproper credit must be given to HTML5 UP if frontend components are used or modified.

## Contributing

Cladevo is an open-source project. Contributions are welcome! If you‚Äôd like to improve the tool, feel free to submit pull requests or report issues.

## Acknowledgments

Cladevo was created to provide an easy-to-use solution for phylogenetic analysis, bridging the gap between complex bioinformatics software and beginners in the field. We appreciate contributions and feedback from the community to enhance its functionality.

Happy analyzing with Cladevo! üöÄ
